pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4214 @@
+
+          MaxObjectCount(:term:`Integer`):
+            Positive integer that defines the maximum number of instances
+            that may be returned from this request.
+
+        Returns:
+
+            A :func:`~py:collections.namedtuple` object containing the following
+            named items:
+
+            * **instances** (:class:`py:list` of :class:`~pywbem.CIMInstance`):
+              Representations of the retrieved instances.
+
+              The `path` attribute of each :class:`~pywbem.CIMInstance`
+              object is `None`, because this operation does not return instance
+              paths.
+
+            * **eos** (:class:`py:bool`):
+              Indicates whether the enumeration session is exhausted after
+              this operation:
+
+              - If `True`, the enumeration session is exhausted, and the
+                server has closed the enumeration session.
+              - If `False`, the enumeration session is not exhausted and the
+                `context` item is the context object for the next operation on
+                the enumeration session.
+
+            * **context** (:func:`py:tuple` of server_context, namespace):
+              A context object identifying the open enumeration session,
+              including its current enumeration state, and the namespace. This
+              object must be supplied with the next pull or close operation for
+              this enumeration session.
+
+              The tuple items are:
+
+              * server_context (:term:`string`):
+                Enumeration context string returned by the server if
+                the session is not exhausted, or `None` otherwise. This string
+                is opaque for the client.
+              * namespace (:term:`string`):
+                Name of the CIM namespace that was used for this operation.
+
+              NOTE: This inner tuple hides the need for a CIM namespace
+              on subsequent operations in the enumeration session. CIM
+              operations always require target namespace, but it never
+              makes sense to specify a different one in subsequent
+              operations on the same enumeration session.
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_ENUMERATION_CONTEXT
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+        """
+        self._test_pull_operations_disabled()
+        return self._pull_response('PullInstances',
+                                   EnumerationContext, MaxObjectCount)
+
+    ####################################################################
+    #
+    #   CloseEnumeration server request
+    #
+    ####################################################################
+
+    def CloseEnumeration(self, EnumerationContext):
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.CloseEnumeration`
+        with data from the instance CIM repository.
```

Comment:
Remove "with data from the instance CIM repository." for consistency with the other operation methods.

Reply:
DONE